lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_download_translations_action.rb

```diff
@@ -0,0 +1,110 @@
+# This action is the new version of android_update_metadata (AndroidUpdateMetadataAction) and should now be used instead of that one
+
+module Fastlane
+  module Actions
+    class AndroidDownloadTranslationsAction < Action
+      def self.run(params)
+        require_relative '../../helper/android/android_localize_helper.rb'
+        require_relative '../../helper/git_helper.rb'
+
+        res_dir = File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', params[:res_dir])
+
+        Fastlane::Helper::Android::LocalizeHelper.create_available_languages_file(
+          res_dir: res_dir,
+          locale_codes: [params[:source_locale]] + params[:locales].map { |h| h[:android] }
+        )
+        Fastlane::Helper::Android::LocalizeHelper.download_from_glotpress(
+          res_dir: res_dir,
+          glotpress_project_url: params[:glotpress_url],
+          glotpress_filters: params[:status_filter].map { |s| { status: s } },
+          locales_map: params[:locales]
+        )
+
+        # Update submodules then lint translations
+        unless params[:lint_task].nil? || params[:lint_task].empty?
+          Fastlane::Helper::GitHelper.update_submodules()
+          Action.sh('./gradlew', params[:lint_task])
+        end
+
+        Fastlane::Helper::GitHelper.commit(message: 'Update translations', files: res_dir, push: true) unless params[:skip_commit]
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Download Android string.xml files from GlotPress and lint the updated translations'
+      end
+
+      def self.details
+        'Download translations from GlotPress, update local strings.xml files accordingly, lint, commit the changes, and push to the remote'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :res_dir,
+            env_name: 'FL_DOWNLOAD_TRANSLATIONS_RES_DIR',
+            description: "The path to the Android project's `res` dir (typically the `<project name>/src/main/res` directory) containing the `values-*` subdirs",
+            type: String,
+            default_value: "#{ENV['PROJECT_NAME']}/src/main/res"
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :glotpress_url,
+            env_name: 'FL_DOWNLOAD_TRANSLATIONS_GLOTPRESS_URL',
+            description: 'URL to the GlotPress project',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :status_filter,
+            env_name: 'FL_DOWNLOAD_TRANSLATIONS_STATUS_FILTER',
+            description: 'The GlotPress status(es) to filter on when downloading the translations',
+            type: Array,
+            default_value: 'current'
```

Comment:
ℹ️ In case you didn't know, `ConfigItem` of `type: Array` automatically accept strings as parameters too (and in such cases split the provided string using `,` as separator to make it into an array).

For more on this, see https://docs.fastlane.tools/advanced/actions/#array-parameters